pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +57,19 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		if _, containsIstioSidecarAnnotation := pod.Annotations["sidecar.istio.io/status"]; !containsIstioSidecarAnnotation || !isPodReady(pod) {
+			continue
+		}
+
+		istioSidecarNames := getIstioSidecarNamesFromAnnotations(pod.Annotations)
```

Comment:
I had only one question if we should really process those annotations.
The name of `istio-proxy` container is the default and it should not change in general.

Reply:
We discussed this issue and decided to process it just in case, although the name will most likely always stay as "istio-proxy". It also takes care of a situation where there are more than one Istio sidecars (although very unlikely)